streams/src/main/java/org/apache/kafka/streams/kstream/JoinWindows.java

```diff
@@ -177,7 +204,9 @@ public long size() {
      * @param afterWindowEnd The grace period to admit out-of-order events to a window.
      * @return this updated builder
      * @throws IllegalArgumentException if the {@code afterWindowEnd} is negative of can't be represented as {@code long milliseconds}
+     * @deprecated since 3.0 Use {@link #ofTimeDifferenceWithNoGrace(Duration)} instead
      */
+    @Deprecated
     public JoinWindows grace(final Duration afterWindowEnd) throws IllegalArgumentException {
```

Comment:
@mjsax I believe those came with your changes in the PR you merged into trunk.

@mjsax, @ableegoldman, please let me know what you think and I can address that as well in the future JIRA ticket.

It could require updating all the unit tests where org.apache.kafka.streams.kstream.JoinWindows.grace() is invoked so that looks like a significant undertaking given today is the cutoff date for merging the feature.

Once you let me know what you would like to do, I will create a JIRA ticket and work on that update, the javadocs and the unit tests.

Thanks.

Reply:
These are the JIRA tickets to track the work to fix the unit tests and javadocs

Improve Javadocs for API Changes from KIP-633
https://issues.apache.org/jira/browse/KAFKA-13021

Migrate all Tests to New API and Remove Suppression for Deprecation Warnings related to KIP-633
https://issues.apache.org/jira/browse/KAFKA-12994

Reply:
Personally I think it makes sense to just disallow calling `ofTimeDifferenceAndGrace(...).grace(...)` entirely, this seems like abusing the API

Reply:
I picked up all the non-testing followup work in this PR so we could try to get it into 3.0: https://github.com/apache/kafka/pull/11114